services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -690,124 +595,56 @@ def _patch_workbench(
-                assert project  # nosec
-                await self.upsert_project_linked_product(
-                    ProjectID(project_uuid), product_name, conn=conn
-                )
-                log.debug(
-                    "DB updated returned row project=%s",
-                    json_dumps(dict(project.items())),
-                )
-                user_email = await self._get_user_email(conn, project.prj_owner)
+    async def get_node_ids_from_project(self, project_uuid: str) -> set[str]:
+        """Returns a set containing all the node_ids from project with project_uuid"""
+        result = set()
+        async with self.engine.acquire() as conn:
+            async for row in conn.execute(
+                sa.select([sa.func.json_object_keys(projects.c.workbench)])
+                .select_from(projects)
+                .where(projects.c.uuid == f"{project_uuid}")
+            ):
+                result.update(row.as_tuple())  # type: ignore
+        return result
 
-                tags = await self._get_tags_by_project(
-                    conn, project_id=project[projects.c.id]
-                )
-                return _convert_to_schema_names(project, user_email, tags=tags)
+    #
+    # Project ACCESS RIGHTS/PERMISSIONS
+    #
 
-    async def delete_user_project(self, user_id: int, project_uuid: str):
-        log.info(
-            "Deleting project with %s for user with %s",
-            f"{project_uuid=}",
-            f"{user_id}",
-        )
+    async def has_permission(
+        self, user_id: UserID, project_uuid: str, permission: Permission
+    ) -> bool:
+        """
+        NOTE: this function should never raise
+        NOTE: if user_id does not exist it is not an issue
+        """
 
         async with self.engine.acquire() as conn:
-            async with conn.begin() as _transaction:
-                project = await self._get_project(
-                    conn, user_id, project_uuid, for_update=True
-                )
-                # if we have delete access we delete the project
-                user_groups: list[RowProxy] = await self.__load_user_groups(
+            try:
+                project = await self._get_project(conn, user_id, project_uuid)
+                user_groups: list[RowProxy] = await self._list_user_groups(
                     conn, user_id
                 )
-                _check_project_permissions(project, user_id, user_groups, "delete")
-                await conn.execute(
-                    # pylint: disable=no-value-for-parameter
-                    projects.delete().where(projects.c.uuid == project_uuid)
-                )
+                check_project_permissions(project, user_id, user_groups, permission)
+                return True
+            except (ProjectInvalidRightsError, ProjectNotFoundError):
+                return False
 
     async def check_delete_project_permission(self, user_id: int, project_uuid: str):
```

Comment:
question: should this method not re-use ```has_permission``` above?

Reply:
The difference is that ``has_permission`` will not raise ``ProjectInvalidRightsError``. This is some extra refactoring we could address next time. I will add a note in #3516